python_modules/dagster/dagster/_core/definitions/op_invocation.py

```diff
@@ -145,6 +153,18 @@ def op_invocation_result(
     if resources_by_param_name:
         context = context.replace_resources(resources_by_param_name)
 
+    config_provided_in_multiple_places = config_input and context.op_config
+    if config_provided_in_multiple_places:
+        raise DagsterInvalidInvocationError("Cannot provide config in both context and kwargs")
+    if config_input:
+        from dagster._config.structured_config import Config
+
+        if isinstance(config_input, Config):
+            config_input = config_input._as_config_dict()  # noqa: SLF001
+        context = context.replace_config(config_input)
```

Comment:
might be nice to inline the instance of check to avoid rewriting `config_arg`/`config_input`